src/server/json_family_test.cc

```diff
@@ -514,4 +514,43 @@ TEST_F(JsonFamilyTest, Del) {
   EXPECT_EQ(resp, R"({})");
 }
 
+TEST_F(JsonFamilyTest, ObjKeys) {
+  string json = R"(
+    {"a":{}, "b":{"a":"a"}, "c":{"a":"a", "b":"bb"}, "d":{"a":1, "b":"b", "c":{"a":3,"b":4}}, "e":1}
+  )";
+
+  auto resp = Run({"set", "json", json});
+  ASSERT_THAT(resp, "OK");
+
+  resp = Run({"JSON.OBJKEYS", "json", "$.a"});
+  EXPECT_THAT(resp, ArgType(RespExpr::NIL_ARRAY));
+
+  resp = Run({"JSON.OBJKEYS", "json", "$.b"});
+  EXPECT_THAT(resp.GetVec(), ElementsAre("a"));
+
+  resp = Run({"JSON.OBJKEYS", "json", "$.*"});
+  ASSERT_THAT(resp, ArrLen(5));
+  const auto& arr = resp.GetVec();
+  EXPECT_THAT(arr[2].GetVec(), ElementsAre("a", "b"));
+  EXPECT_THAT(arr[3].GetVec(), ElementsAre("a", "b", "c"));
```

Comment:
just to confirm OBJKEYS never goes beyond 2 levels in elasticache?

Reply:
It depends on what JsonPath you supply to the command.  I used JsonPath($.*) which traverses only one level of the hierarchy.
I added a UT that demonstrates recursive descent traversal ($..*) from level 2.

Reply:
I checked that my implementation is aligned with Elasticache and RedisJson. those implementations have some differences, one of them is: Elasticache prints an empty object and an item that is not an object as an empty array, and RedisJson differs between an empty object and an item that is not an object, an empty object is printed as an empty array and not object is printed as null.

Reply:
which flavor we are aligned with?

Reply:
We are aligned with Elasticache